server/data/resolvers.js

```diff
@@ -174,50 +171,28 @@ const resolvers = {
         }
       })
     },
-    addVote: (obj, args) => {
-      User.findOne({account: args.voter}).exec((err, user) => {
-        if (err) {
-          console.error(err)
-        } else {
-          Project.findOne({address: args.projectAddress}).exec((err, project) => {
-            if (err) { console.error(err) }
-            Task.findOne({project: project.id, index: args.taskIndex}).exec((err, task) => {
-              if (err) { console.error(err) }
-              let userVoteObj = new VoteRecord({
-                _id: new mongoose.Types.ObjectId(),
-                amount: args.amount,
-                pollID: args.pollID,
-                project: args.projectAddress,
-                revealed: false,
-                rescued: false,
-                salt: args.salt,
-                task: task.id,
-                type: args.type,
-                vote: args.vote,
-                voter: user.id
-              })
-              let index = _.sortedIndexBy(user.voteRecords, userVoteObj, (o) => o.amount)
-              user.voteRecords.splice(index, 0, userVoteObj)
-              user.markModified('voteRecords')
-              user.save(err => {
-                if (err) return console.log(err)
-                return userVoteObj
-              })
-            })
-          })
-
-          // userVoteObj.save((err, vote) => {
-          //   if (err) return console.log(err)
-          //   let index = _.sortedIndexBy(user.voteRecords, vote, (o) => o.amount)
-          //   user.voteRecords.splice(index, 0, vote)
-          //   user.markModified('voteRecords')
-          //   user.save((err) => {
-          //     if (err) return console.log(err)
-          //     return vote
-          //   })
-          // })
-        }
-      })
+    addVote: async (obj, args) => {
+      try {
+        const user = await User.findOne({account: args.voter})
+        const project = await Project.findOne({account: args.projectAddress})
+        const task = await Task.findOne({project: project.id, index: args.taskIndex})
+        const vote = await Vote.findOneAndUpdate({
+          amount: args.amount,
+          pollID: args.pollID,
+          project: project.id,
+          revealed: false,
+          rescued: false,
+          task: task.id,
+          type: args.type,
+          voter: user.id
+        }, { $set: {
+          salt: args.salt,
+          vote: args.vote
+        }}, {upsert: true})
+        if (!vote) console.error('vote not added successfully')
+      } catch (err) {
+        console.error('Error Adding Vote')
```

Comment:
Wonder if you want to rethrow here, so it bubbles up to what gets returned by graphql?  What does this mutation return?